pgserver/stmt.go

```diff
@@ -271,18 +271,121 @@ var (
 // get the regex to match any table in pg_catalog in the query.
 func getPgCatalogRegex() *regexp.Regexp {
 	initPgCatalogRegex.Do(func() {
-		var tableNames []string
+		var internalNames []string
 		for _, table := range catalog.GetInternalTables() {
 			if table.Schema != "__sys__" {
 				continue
 			}
-			tableNames = append(tableNames, table.Name)
+			internalNames = append(internalNames, table.Name)
 		}
-		pgCatalogRegex = regexp.MustCompile(`(?i)\b(FROM|JOIN|INTO)\s+(?:pg_catalog\.)?(` + strings.Join(tableNames, "|") + `)`)
+		for _, view := range catalog.InternalViews {
+			if view.Schema != "__sys__" {
+				continue
+			}
+			internalNames = append(internalNames, view.Name)
+		}
+		pgCatalogRegex = regexp.MustCompile(
+			`(?i)\b(FROM|JOIN|INTO)\s+(?:pg_catalog\.)?(?:"?(` + strings.Join(internalNames, "|") + `)"?)`)
 	})
 	return pgCatalogRegex
 }
 
 func ConvertToSys(sql string) string {
 	return getPgCatalogRegex().ReplaceAllString(RemoveComments(sql), "$1 __sys__.$2")
 }
+
+// The below code lines are generated by Claude Sonnet 3.5 and ChatGPT o1 preview
+var (
+	renameMacroRegex     *regexp.Regexp
+	initRenameMacroRegex sync.Once
+	macroRegex           *regexp.Regexp
+	initMacroRegex       sync.Once
+)
+
+func getRenamePgCatalogFuncRegex() *regexp.Regexp {
+	initRenameMacroRegex.Do(func() {
+		var internalNames []string
+		for _, view := range catalog.InternalMacros {
+			if strings.ToLower(view.Schema) != "pg_catalog" {
+				continue
+			}
+			// Quote the function name to ensure safe regex usage
+			internalNames = append(internalNames, regexp.QuoteMeta(view.Name))
+		}
+
+		namesAlt := strings.Join(internalNames, "|")
+
+		// Compile the regex
+		// The pattern matches:
+		// - Branch A: "pg_catalog.<funcName>("
+		// - Branch B: "<funcName>(" without a preceding "."
+		pattern := `(?i)(?:pg_catalog\.("?(?:` + namesAlt + `)"?)\(|(^|[^\.])("?(?:` + namesAlt + `)"?)\()`
+		renameMacroRegex = regexp.MustCompile(pattern)
+	})
+	return renameMacroRegex
+}
+
+func ConvertPgCatalogFuncToSys(sql string) string {
```

Comment:
It would be better to add comments for these functions to explain what they are doing.